Broken dependency when using OpenTelemetry + REST Client on main

## Describe the bug
When combining `quarkus-opentelemetry-exporter-otlp` and `rest-client` extensions, Maven is unable to resolve the `io.quarkus:quarkus-opentelemetry-rest-client-deployment` dependency unless it is locally cached.  

### Expected behavior
Build success

### Actual behavior
```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:generate-code (default) on project opentelemetry-test: Quarkus code generation phase has failed: Failed to bootstrap the application: Failed to create the application model for org.acme:opentelemetry-test::jar:1.0.0-SNAPSHOT: Failed to resolve artifacts: Could not find artifact io.quarkus:quarkus-opentelemetry-rest-client-deployment:jar:999-SNAPSHOT
```

## To Reproduce
Make sure you do not have `io.quarkus:quarkus-opentelemetry-rest-client-deployment:jar:999-SNAPSHOT` in local Maven repo and a remote repo with Quarkus `999-SNAPSHOT` is available.
```
mvn io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:create \
    -DplatformArtifactId=quarkus-bom \
    -DprojectGroupId=org.acme \
    -DprojectArtifactId=opentelemetry-test \
    -Dextensions="rest-client,quarkus-opentelemetry-exporter-otlp"
cd opentelemetry-test
mvn clean compile
```

See also https://github.com/quarkusio/quarkus-quickstarts/tree/development/opentelemetry-quickstart, which fails in the same manner when using a remote repository.

## Environment:
```
Apache Maven 3.8.1
Java version: 11.0.10, vendor: AdoptOpenJDK
Quarkus main (999-SNAPSHOT)
```

--- a/independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/DeploymentInjectingDependencyVisitor.java
+++ b/independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/DeploymentInjectingDependencyVisitor.java
@@ -546,7 +546,8 @@ public class DeploymentInjectingDependencyVisitor {
             collectingTopRuntimeNodes = false;
             final ExtensionDependency extDep = getExtensionDependency();
             final DependencyNode originalNode = collectDependencies(info.runtimeArtifact, extDep.exclusions);
-            final DependencyNode rtNode = extDep.runtimeNode;
+            final DefaultDependencyNode rtNode = (DefaultDependencyNode) extDep.runtimeNode;
+            rtNode.setRepositories(originalNode.getRepositories());
             // if this node has conditional dependencies on its own, they may have been activated by this time
             // in which case they would be included into its children
             List<DependencyNode> currentChildren = rtNode.getChildren();

